### PR TITLE
Added ConfigService, support mutability of static xcm configs

### PR DESCRIPTION
**Pull request by nohaapav**
### Description

Introducing ConfigService / IConfigService to allow seamless custom xcm registry configuration, with default fallback to static, existing one. For details visit [here](https://github.com/PureStake/xcm-sdk/issues/122)

### Checklist

- [x] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/PureStake/moonbeam-docs) repository.
- [x] If this requires it, I have updated the Readme
- [x] If necessary, I have updated the examples
- [x] I have verified if I need to create/update unit tests
- [x] I have verified if I need to create/update acceptance tests
- [x] If necessary, I have run acceptance tests on this branch in CI

### CODE REVIEW

**Comment by nohaapav** [comment]
@ekenigs Hi so PR has landed :)

I've created `MutableConfigService` that should cover basic user needs:

- `updateAssets(asset: Asset): void` - Create new or update existing asset
- `updateChains(chain: AnyChain): void` - Create new or update existing chain
- `updateChain(chain: AnyChain, chainDelta: Partial<AnyChain>): void` - Update existing chain delta
- `updateChainConfig(chainConfig: ChainConfig): void` - Create new or update existing chain config
- `updateAssetConfig(chain: AnyChain, assetConfig: AssetConfig): void` - Create new or update existing chain asset config

Current static config builder utils were moved to ConfigService which acts as fallback for `ConfigBuilder` & `Sdk` unless user specify otherwise.

**Comment by ekenigs** [comment]
@nohaapav Hey, thanks for the PR. I hope that I will have some time next week to review it.

**Comment by nohaapav** [comment]
> @nohaapav Hey, thanks for the PR. I hope that I will have some time next week to review it.

No worries, i've published custom @org/xcm-sdk & @org/xcm-config packages so it's not a blocker for me :) ..

**Comment by nohaapav** [comment]
> @nohaapav Hey, thanks for the PR. I hope that I will have some time next week to review it.

@ekenigs Just pls let me know if it make sense to get PR merged in reasonable time in general. I can just update custom fork with fixes, it consumes less of my time as well. I know u are busy, so am i so let's find win win situation here :)

**Comment by ekenigs** [comment]
> > @nohaapav Hey, thanks for the PR. I hope that I will have some time next week to review it.
> 
> @ekenigs Just pls let me know if it make sense to get PR merged in reasonable time in general. I can just update custom fork with fixes, it consumes less of my time as well. I know u are busy, so am i so let's find win win scenario here :)

@nohaapav
Unfortunately, your timing coincided with a particularly busy period for us, as we're in the final stages of a new project. Due to this, I might not be able to review your PR immediately, but I'll do my best to take a look at it by tomorrow.

File: packages/config/src/ConfigService/MutableConfigService.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  EvmParachain,
+  EvmParachainConstructorParams,
+  Parachain,
+  ParachainConstructorParams,
+} from '@moonbeam-network/xcm-types';
+import { AssetConfig } from '../types/AssetConfig';
+import { ChainConfig } from '../types/ChainConfig';
+import { ConfigService } from './ConfigService';
+
+export class MutableConfigService extends ConfigService {
```
Comment (ekenigs) [inline packages/config/src/ConfigService/MutableConfigService.ts:13]: I really like that you created `ConfigService`. I'm just not sure about `MutableConfigService`. Maybe I would just add `updateAssets` (which I would rename to `updateAsset`), `updateChains` (`updateChain`), `updateChainConfig` to `ConfigService`. Other methods I wouldn't add.
If users need to change a few `chains`, `assets`, `chainConfigs` that should be enough  and if they need to more they can always create a new instance of `ConfigService`. Or you need them?

**Review by ekenigs — CHANGES_REQUESTED** [review]
@nohaapav thanks again for your PR. Please check my comment and also please add a changeset (`npm run changeset`) for both Config and Sdk as a minor.

File: packages/config/src/ConfigService/MutableConfigService.ts (hunk 1 above)
Comment (nohaapav) [reply packages/config/src/ConfigService/MutableConfigService.ts:13]: @ekenigs We can drop MutableConfigService :) .. You're right, user can always inherit from ConfigService & write 'extra' utility functions on top of that. I don't really use them, frankly i ended u with following:

```
function configure(configService: MutableConfigService) {
  assets.forEach((asset) => {
    configService.updateAssets(asset);
  });

  chains.forEach((chain) => {
    configService.updateChains(chain);
  });

  chainsConfig.forEach((config) => {
    configService.updateChainConfig(config);
  });
}
```
It's was more like nice to have for new folks to ease their lives :) I'm just thinking maybe one more function `reset()` to wipe the entire config if you want to setup the configuration from scratch .. Reset can be facade calling map.clear() for both chains, assets, chainConfigs. WDYT ?

Comment (nohaapav) [reply packages/config/src/ConfigService/MutableConfigService.ts]: ;) Alright .. Should be done. Changeset updated.

**Review by ekenigs — APPROVED** [review]

**Pull request merged**